main/guides/zoe/contract-details.md

```diff
@@ -0,0 +1,121 @@
+# Durable Contract Details
+
+Zoe allows you to write smart contracts that manage interactions between cooperative, but possibly untrusting,
+parties. Some contracts perform a single simple task like trading one good for another, and then are
+done. Others stay around and manage assets and interactions over time. The first kind don't need persistence
+or the ability to upgrade, while the second kind need to make use of more sophisticated functionality.
+
+When a contract is intended to continue running and serving many customers over a long time, its objects and
+data need to be persistent, its owners or managers may need to be able to adjust parameters or perform other
+governance actions, and it needs to be upgradeable so that the code can adapt over time.
+
+## Durable Objects
+
+The first step toward contracts that can be upgraded is storing all the data that a later incarnation will
+need. This means putting relevant state in
+[Baggage](http://localhost:5173/guides/zoe/contract-upgrade.html#baggage), and ensuring that reachable objects
+that will be accessible to users have an identity that can be maintained as the behavior changes with contract
+upgrades.
+
+We use zone.exo(), zone.exoClass(), and zone.exoClassKit() to define durable objects.
+
+[Zone](http://localhost:5173/glossary/#zone) provides an interface for defining objects and classes that
+supports both ephemeral objects (allocated on the heap), and durable objects that can persist and that
+SwingSet will page in or out on demand.
```

Comment:
Is "SwingSet" in the reader's vocabulary at this point? Link to glossary entry?

Reply:
done